### ISSUE TITLE
Need a means to pass TLS credentials to WebSocket

### ISSUE THREAD

**Issue by chelliwell** [graphql-dotnet/graphql-client#118]
> I think, at a first quick look (cos I gotta get home soon!) - it maybe needs being able to get some ClientWebSocketOptions in, for instance, GraphQL.Client.Http.GraphQLHttpWebSocket.InitializeWebSocket() ...?
> [I'm not familiar with .NET WebSockets]

Confirmed. If I add 
`nativeWebSocket.Options.ClientCertificates.Add(new X509Certificate2("My-cert-and-key.pfx", "mypassword"));`
then the authentication succeeds. I think the websockets layers therefore need a way for the app to be able to pass these options down? (Possibly 'UseDefaultCredentials' too)

_Originally posted by @chelliwell in https://github.com/graphql-dotnet/graphql-client/issues/114#issuecomment-492623214_

**Comment by chelliwell** [comment]
Hopefully this can be addressed fairly soon. Or - though C# isn't much of a strength of mine (but I have colleagues who are more expert) - any suggestions on the cleanest ways/places to add this ability?

**Comment by chelliwell** [comment]
I believe I've solved this suitably in GraphQL.Client.Http.GraphQLHttpWebSocket.InitializeWebSocket():
```
				switch (clientWebSocket)
				{
					case ClientWebSocket nativeWebSocket:
						nativeWebSocket.Options.AddSubProtocol("graphql-ws");
						nativeWebSocket.Options.ClientCertificates = ((HttpClientHandler)(_options.HttpMessageHandler)).ClientCertificates;
						nativeWebSocket.Options.UseDefaultCredentials = ((HttpClientHandler)(_options.HttpMessageHandler)).UseDefaultCredentials;
						break;
					case System.Net.WebSockets.Managed.ClientWebSocket managedWebSocket:
						managedWebSocket.Options.AddSubProtocol("graphql-ws");
						managedWebSocket.Options.ClientCertificates = ((HttpClientHandler)(_options.HttpMessageHandler)).ClientCertificates;
						managedWebSocket.Options.UseDefaultCredentials = ((HttpClientHandler)(_options.HttpMessageHandler)).UseDefaultCredentials;
						break;
					default:
						throw new NotSupportedException($"unknown websocket type {clientWebSocket.GetType().Name}");
				}
```
Together with 'using System.Net.Http;'
Could this be reviews and added in to the relevant branch(es) please?

**Comment by rose-a** [comment]
fixed in #152
